src/pytorch_lightning/utilities/cloud_io.py

```diff
@@ -25,6 +28,26 @@
 from pytorch_lightning.utilities.types import _PATH
 
 
+class _ThreadQueue(threading.Thread):
```

Comment:
Let's use ThreadPoolExecutor, we don't need to reinvent the wheel :)

In the plugin do:
`executor = ThreadPoolExecutor(1)`

in save_checkpoint:
`executor.submit(func, args)`

Reply:
But I need to join the queue and thread properly, can't find a way to do that using `ThreadPoolExecutor`.

Reply:
You don't need to, because it will do so when the task finishes.

Reply:
will it make sure that the thread will run continuously until the queue is empty?

Reply:
Yes, as long as there are items on the queue, the workers will do work.
https://superfastpython.com/threadpoolexecutor-in-python/#How_Does_ThreadPoolExecutor_Work_Internally